diff --git a/homeassistant/components/elkm1/__init__.py b/homeassistant/components/elkm1/__init__.py
index 946e40b7e2..3be2263bcd 100644
--- a/homeassistant/components/elkm1/__init__.py
+++ b/homeassistant/components/elkm1/__init__.py
@@ -15,6 +15,7 @@ from homeassistant.const import (
     CONF_PASSWORD,
     CONF_TEMPERATURE_UNIT,
     CONF_USERNAME,
+    CONF_CODE,
     TEMP_CELSIUS,
     TEMP_FAHRENHEIT,
 )
@@ -122,6 +123,7 @@ DEVICE_SCHEMA = vol.Schema(
         vol.Optional(CONF_PREFIX, default=""): vol.All(cv.string, vol.Lower),
         vol.Optional(CONF_USERNAME, default=""): cv.string,
         vol.Optional(CONF_PASSWORD, default=""): cv.string,
+        vol.Optional(CONF_CODE, default=""): cv.string,
         vol.Optional(CONF_AUTO_CONFIGURE, default=False): cv.boolean,
         # cv.temperature_unit will mutate 'C' -> '°C' and 'F' -> '°F'
         vol.Optional(
@@ -203,6 +205,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     if not conf[CONF_AUTO_CONFIGURE]:
         # With elkm1-lib==0.7.16 and later auto configure is available
         config["panel"] = {"enabled": True, "included": [True]}
+        config["code"] = conf[CONF_CODE]
         for item, max_ in ELK_ELEMENTS.items():
             config[item] = {
                 "enabled": conf[item][CONF_ENABLED],
@@ -355,6 +358,7 @@ class ElkEntity(Entity):
         self._element = element
         self._prefix = elk_data["prefix"]
         self._temperature_unit = elk_data["config"]["temperature_unit"]
+        self._code = elk_data["config"]["code"]
         # unique_id starts with elkm1_ iff there is no prefix
         # it starts with elkm1m_{prefix} iff there is a prefix
         # this is to avoid a conflict between
diff --git a/homeassistant/components/elkm1/alarm_control_panel.py b/homeassistant/components/elkm1/alarm_control_panel.py
index 1c299e6880..e1809aaad1 100644
--- a/homeassistant/components/elkm1/alarm_control_panel.py
+++ b/homeassistant/components/elkm1/alarm_control_panel.py
@@ -201,6 +201,8 @@ class ElkArea(ElkAttachedEntity, AlarmControlPanelEntity, RestoreEntity):
         attrs[ATTR_CHANGED_BY_KEYPAD] = self._changed_by_keypad
         attrs[ATTR_CHANGED_BY_TIME] = self._changed_by_time
         attrs[ATTR_CHANGED_BY_ID] = self._changed_by_id
+        if self._code != "":
+            attrs["code_arm_required"] = False
         return attrs
 
     @property
@@ -242,26 +244,38 @@ class ElkArea(ElkAttachedEntity, AlarmControlPanelEntity, RestoreEntity):
 
     async def async_alarm_arm_home(self, code=None):
         """Send arm home command."""
+        if code is None:
+            code = self._code
         self._element.arm(ArmLevel.ARMED_STAY.value, int(code))
 
     async def async_alarm_arm_away(self, code=None):
         """Send arm away command."""
+        if code is None:
+            code = self._code
         self._element.arm(ArmLevel.ARMED_AWAY.value, int(code))
 
     async def async_alarm_arm_night(self, code=None):
         """Send arm night command."""
+        if code is None:
+            code = self._code
         self._element.arm(ArmLevel.ARMED_NIGHT.value, int(code))
 
     async def async_alarm_arm_home_instant(self, code=None):
         """Send arm stay instant command."""
+        if code is None:
+            code = self._code
         self._element.arm(ArmLevel.ARMED_STAY_INSTANT.value, int(code))
 
     async def async_alarm_arm_night_instant(self, code=None):
         """Send arm night instant command."""
+        if code is None:
+            code = self._code
         self._element.arm(ArmLevel.ARMED_NIGHT_INSTANT.value, int(code))
 
     async def async_alarm_arm_vacation(self, code=None):
         """Send arm vacation command."""
+        if code is None:
+            code = self._code
         self._element.arm(ArmLevel.ARMED_VACATION.value, int(code))
 
     async def async_display_message(self, clear, beep, timeout, line1, line2):
